app/jobs/sync_pardot_prospects_and_subscriptions_job.rb

```diff
@@ -0,0 +1,46 @@
+class SyncPardotProspectsAndSubscriptionsJob < ApplicationJob
+  queue_as :default
+  sidekiq_options retry: 2
+
+  def perform(salesforce_ids = [])
+    # without an argument, this processes all contacts - else you can specify an array of salesforce_ids to process
+    query = Contact.all
+    query = query.where(salesforce_id: salesforce_ids) unless salesforce_ids.empty?
+
+    query.each do |contact|
+      current_subscriptions = Subscription.includes(:list).where(contact: contact)
+      new_subscriptions = []
+
+      prospect = Pardot.client.prospects.read_by_fid(contact.salesforce_id)
+      next unless prospect['lists']
+
+      prospect['lists']['list_subscription'].each do |subscription|
+        if subscription.include?('list')
+          # Returns an array if it only has one subscription, but a hash otherwise. O_o This seems to handle it nicely.
+          subscription = if subscription.is_a?(Array)
+                           subscription[1]
+                         else
+                           subscription['list']
+                         end
```

Comment:
at this point is the `subscription` variable now really representing a list?  If so that makes the naming confusing below, like `new_subscriptions.append(subscription['id'].to_i)` makes it sound like `subscription['id']` is a ... subscription ID but then the next line `List.where(pardot_id: subscription['id'])` makes it sound like that ID is a list ID.

Reply:
I see how it can be confusing. `subscription` in this context is a subscription item for the user from Pardot. I'll try to clean this up and make it more clear.